# Case Study on Measures of Central Tendency and Dispersion


Main objectives of the case study are as follows.

1. Find out how many males and females participated in the test.
2. What do you think about the students' parental level of education?
3. Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course
4. What do you think about the scoring variation for math, reading and writing
based on
● Gender
● Test preparation course
5. The management needs your help to give bonus points to the top 25% of
students based on their maths score, so how will you help the management
to achieve this.

In [1]:
import pandas as pd 
import numpy as np

In [2]:
dff=pd.read_csv('D:\DSA\StudentsPerformance.csv')
dff

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

# 1. Find out how many males and females participated in the test.


In [3]:
dff['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

INSIGHTS:There are 518 females and 482 males participated in the test.

# 2. What do you think about the students' parental level of education?

In [5]:
dff['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

Insights: Most of the parents have some college education.59 parents have masters degree.

# 3. Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course

In [6]:
#Based on Gender.

In [7]:
dff3=dff.groupby('gender')['math score'].mean()
dff3

gender
female    63.376448
male      68.821577
Name: math score, dtype: float64

In [8]:
dff3=dff.groupby('gender')['reading score'].mean()
dff3

gender
female    72.590734
male      65.545643
Name: reading score, dtype: float64

In [9]:
dff3=dff.groupby('gender')['writing score'].mean()
dff3

gender
female    72.467181
male      63.446058
Name: writing score, dtype: float64

INSIGHTS: Based on Gender males score highest for Maths and females score highest for reading and writing

In [10]:
# Based on Test preparation course.

In [11]:
dff3=dff.groupby('test preparation course')['math score'].mean()
dff3

test preparation course
completed    69.96648
none         63.78972
Name: math score, dtype: float64

In [12]:
dff3=dff.groupby('test preparation course')['reading score'].mean()
dff3

test preparation course
completed    74.175978
none         66.417445
Name: reading score, dtype: float64

In [13]:
dff3=dff.groupby('test preparation course')['writing score'].mean()
dff3

test preparation course
completed    74.684358
none         64.457944
Name: writing score, dtype: float64

INSIGHTS:On an average those who completed test preparation course scored more for maths, reading and writing.

# 4. Scoring variation for math, reading and writing based on
● Gender
● Test preparation course

In [14]:
#Based on Gender.

In [17]:
dff3=dff.groupby('gender')['math score'].var()
dff3

gender
female    256.958593
male      211.889097
Name: math score, dtype: float64

In [18]:
dff3=dff.groupby('gender')['writing score'].var()
dff3

gender
female    220.369327
male      202.413924
Name: writing score, dtype: float64

In [19]:
dff3=dff.groupby('gender')['reading score'].var()
dff3

gender
female    207.677438
male      200.211010
Name: reading score, dtype: float64

INSIGHTS: Scoring variation is high for females for math, reading and writing.

In [21]:
#Based on test preparation course.

In [22]:
dff3=dff.groupby('test preparation course')['math score'].var()
dff3

test preparation course
completed    210.884027
none         246.668662
Name: math score, dtype: float64

In [23]:
dff3=dff.groupby('test preparation course')['reading score'].var()
dff3

test preparation course
completed    183.265864
none         213.419851
Name: reading score, dtype: float64

In [24]:
dff3=dff.groupby('test preparation course')['writing score'].var()
dff3

test preparation course
completed    175.202612
none         226.251739
Name: writing score, dtype: float64

INSIGHTS:Those who have completed the test preparation course has less variance from the mean and for those who have not taken any course has higher variance

# 5. The management needs your help to give bonus points to the top 25% of
students based on their maths score.

In [5]:
pd.qcut(dff['math score'],q=4).value_counts()

(66.0, 77.0]      266
(-0.001, 57.0]    265
(57.0, 66.0]      238
(77.0, 100.0]     231
Name: math score, dtype: int64

Insights:Top 25% students have scores above 77, so the management can give bonus point for students having scores more than 77. and there are 231 students in top 25%.